docs/framework/chat-assistant.mdx

```diff
@@ -2,153 +2,325 @@
+  # More documentation is available at https://dev.writer.com/framework
+
+
+  # Initialise the state
+  wf.init_state({
+      "my_app": {
+          "title": "AI STARTER"
+      },
+  })
+  ```
+
+  ...and replace it with the following:
+
+  ```python
+  # Initialize the state
+  wf.init_state({
+      "conversation": writer.ai.Conversation(),
+      "my_app": {
+          "title": "CHAT ASSISTANT"
+      },
+  })
+  ```
+
+  The Writer Framework's `init_state()` method sets the initial value of `state`, a dictionary containing values that define the state of the application. The key-value pairs in `state` are how you store values used by your app and how you pass data between the back-end code and the UI.
+
+  The code above sets the initial value of `state` so that it has two key-value pairs:
+
+  - `conversation`: An object that keeps a record of the conversation that the user is having with the LLM. You'll bind its value to the **Chatbot** component soon.
+  - `my_app`: A dictionary containing values that define the application's appearance. This dictionary has a single key-value pair, `title`, which defines the text that appears as the application's title in the **Header**.
+  </Step>
+
+  <Step title="Create a handler to respond to the user's input">
+  The application needs a function to handle incoming chat messages. Add the following to the end of `main.py`:
+
+  ```python
+  def generate_completion(state, payload):
+    print(f"Here's what the user entered: {payload['content']}")
+    state["conversation"] += payload
+    print(f"Conversation: {state['conversation'].messages}")
+    try:
+        for index, chunk in enumerate(state["conversation"].stream_complete()):
+            print(f"Chunk {index}: {chunk}")
+            if not chunk.get("content"):
+                chunk["content"] = ""
             state["conversation"] += chunk
+            
+        print(f"state['conversation']:\n{state['conversation'].messages}")
+    except Exception as e:
+        print(f"Error during stream_complete: {e}")
     ```
 
-    This code uses the streaming function of the `Conversation` method, which is a wrapper for the `chat` API endpoint. Each chunk returned from the stream is added to the `conversation` variable in the application state.
+    The `generate_completion()` function will be called when the user enters a prompt, which is contained in the `payload` object. The `payload` object is  added to the `conversation` object contained in the application's `state`, which adds the user's prompt to the record of the conversation between the user and the LLM.
+
+    After adding the user's prompt to the conversational record, `generate_completion()` calls the `conversation` object's `stream_complete()` method, which generates an LLM completion based on the conversation so far. As its name implies, `stream_complete()` returns the completion as a stream of text chunks, which are captured and added to the `conversation` object.
```

Comment:
Let's link to the AI module guide at the end of this if people want to learn more, for example if they want to learn more.